### PR TITLE
#130 Added module name configuration in Jar file

### PR DESCRIPTION
**Pull request by jiakuan**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -45,6 +45,8 @@
         <project.build.outputEncoding>UTF-8</project.build.outputEncoding>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.project.outputEncoding>UTF-8</project.project.outputEncoding>
+
+        <jopt.simple.module.name>jopt.simple</jopt.simple.module.name>
```
Comment (twoqubed) [inline pom.xml]: Also, what about joptsimple instead? To me, the . implies some sort of namespace.

File: pom.xml [hunk 2]
```diff
@@ -112,6 +114,24 @@
...
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>test-jar</goal>
+                        </goals>
```
Comment (twoqubed) [inline pom.xml]: What does this result in? An `Automatic-Module-Name` in the manifest? If so, why is being applied in the test-jar goal?

File: pom.xml (hunk 1 above)
Comment (jiakuan) [reply pom.xml]: That's a good idea. Have changed to `joptsimple`

File: pom.xml (hunk 2 above)
Comment (jiakuan) [reply pom.xml]: Yes, just explicitly set a module name in the manifest. Seems `test-jar` goal is not needed, although I did check the result jar had the module name. Now I've removed the `test-jar` goal from it.

File: pom.xml (hunk 1 above)
Comment (jiakuan) [reply pom.xml]: If we didn't specify a module name, Java 9 would automatically generate a module name `jopt.simple` from the jar name `jopt-simple`.  So with this better name `joptsimple` specified, Java 9 will be able to recognise `joptsimple`.

**Pull request merged**